sim/warrior/stances.go

```diff
@@ -105,86 +103,40 @@ func (warrior *Warrior) registerDefensiveStanceAura() {
 			Duration: core.NeverExpires,
 			OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
 				if result.Outcome.Matches(core.OutcomeBlock | core.OutcomeDodge | core.OutcomeParry) {
-					if sim.RandomFloat("Enrage Trigger Chance") <= 0.5*float64(warrior.Talents.ImprovedDefensiveStance) {
+					if sim.RandomFloat("Enrage Trigger Chance") <= 0.1*float64(warrior.Talents.BastionOfDefense) {
```

Comment:
You should cache any values that dont depend on dynamic values outside of the callbacks.